index.js

```diff
@@ -11,28 +10,136 @@
+      cell.y = y;
 
       //this function adds the click event to each cell
-      cell.addEventListener( 'click', (e) => {
-          let cell = e.target; // get the element clicked
-          console.log( cell.textContent) //display the coordinates in the console
-          cell.style.visibility = 'hidden';// this  means that the contents of the element will be invisible, but the element stays in its original position and size / try it clicking on any of the black cells (in your browser) and see whats happens
-          //cell.style.background ="purple"; //with this propertie you can change the background color of the clicked cell. try comment the line bellow and uncomment this line. Do not forget to save this file and refresh the borwser to see the changes
+      cell.addEventListener("click", (e) => {
+        if (player.playing) {
+          return;
+        }
+
+        let cell = e.target; // get the element clicked
+        console.log(cell.textContent); //display the coordinates in the console
+        //cell.style.visibility = "hidden"; // this  means that the contents of the element will be invisible, but the element stays in its original position and size / try it clicking on any of the black cells (in your browser) and see whats happens
+
+        // cell.style.background = "purple"; //with this propertie you can change the background color of the clicked cell. try comment the line bellow and uncomment this line. Do not forget to save this file and refresh the borwser to see the changes
+
+        //strike opponent's ships
+        if (e.target.value == 1) {
+          cell.style.visibility = "hidden";
+          player.shipCount--;
+          player.lives--;
+          alert("You hit a ship!");
+        } else if (e.target.value == 0) {
+          cell.style.background = "purple";
+          alert("It's a miss!");
+        }
+
+        let updatedLives = document.querySelector(`#ships_player${id}`);
+        updatedLives.textContent = player.lives;
+
+        //taking turns
+        player.playing = true;
+        opponent.playing = false;
+
+        const turn = document.querySelector("#turn_player");
+        turn.innerHTML = opponent.name;
+
+        console.log(e.target.value);
+
+        //display the winner
+        if (player1.lives == 0 && player2.lives > 0) {
+          alert(`${player2.name} you won the game!`);
+          alert("Loading a new game.");
+          location.reload();
+          return `The winner is...${player2.name}!`;
+        } else if (player2.lives == 0 && player1.lives > 0) {
+          alert(`${player1.name} you won the game!`);
+          alert("Loading a new game.");
+          location.reload();
+          return `The winner is...${player1.name}!`;
+        }
       });
 
       li.appendChild(cell); //adding each cell into the row number x
     }
 
-     board_Player1.appendChild(li); //adding each row into the board
-}
+    board.appendChild(li); //adding each row into the board
+  }
```

Comment:
This block is too long, that makes the code inside difficult to read. It would be better having different functions that we can call inside the loop to make it more modular.